solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.4;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../token/TBTCToken.sol";
+import "../token/IReceiveApproval.sol";
+import "../GovernanceUtils.sol";
+
+/// @title TBTC v2 Vending Machine
+/// @notice The Vending Machine is the owner of TBTC v2 token and can mint
+///         TBTC v2 tokens in 1:1 ratio from TBTC v1 tokens. TBTC v2 can be
+///         unminted back to TBTC v1 with or without a fee - fee parameter is
+///         controlled by the governance. This implementation acts as a bridge
+///         between TBTC v1 and TBTC v2 token, allowing to mint TBTC v2 before
+///         the system is ready and fully operational without sacrificing any
+///         security guarantees and decentralization of the project.
+///         Vending Machine as a contract itself is not upgradeable, though
+///         TBTC v2 token ownership can be updated in a two-step,
+///         governance-controlled process. It is expected that this process
+///         will be executed before the v2 system launch.
+contract VendingMachine is Ownable, IReceiveApproval {
+    using SafeERC20 for IERC20;
+    using SafeERC20 for TBTCToken;
+
+    /// @notice The time delay that needs to pass between initializing and
+    ///         finalizing update of any governable parameter in this contract.
+    uint256 public constant GOVERNANCE_DELAY = 48 hours;
+
+    /// @notice This divisor for precision purposes. Used to represent fractions
+    ///         in parameter values.
+    uint256 public constant FLOATING_POINT_DIVISOR = 1e18;
+
+    IERC20 public immutable tbtcV1;
+    TBTCToken public immutable tbtcV2;
+
+    /// @notice The fee for unminting TBTC v2 back into TBTC v1. The fee is
+    ///         a portion of the amount being unminted multiplied by
```

Comment:
```suggestion
    ///         a portion of the amount being unminted divided by
```

Reply:
`unmintFee` is a decimal in 1e18 precision:
```
    1 => 10^18 = 1000000000000000000
  0.1 => 10^17 = 100000000000000000
 0.01 => 10^16 = 10000000000000000
0.001 => 10^15 = 1000000000000000
(...)
```

If we expect the fee to be 1% of the amount being unminted, then `unmintFee` should be set to 0.01 in 1e18 precision which is 10000000000000000. To calculate the unmint fee value we do:
```
function unmintFeeFor(uint256 amount) public view returns (uint256) {
  return (amount * unmintFee) / FLOATING_POINT_DIVISOR;
}
```
We *multiply* and the division is only to eliminate 1e18 precision multiplier from the final value.

I do not think `unmintFee` is a percentage because if `unmintFee` is set to `10000000000000000`, it is `0.01` and not `0.01%`. If we were to say it is a percentage, then the value of `unmintFee` for this case should be `1e18` to represent `1%` but this would complicate `unmintFeeFor` as we'd have to additionally divide by `100`.

The alternative proposition of this doc:
```
    /// @notice The fee for unminting TBTC v2 back into TBTC v1 represented as
    ///         1e18 precision fraction. The fee is proportional to the amount
    ///         being unminted and added on the top of the amount being unminted.
    ///         To calculate the fee value, the amount being unminted needs
    ///         to be multiplied by `unmintFee` and divided by 1e18.
    ///         For example, `unmintFee` set to 1000000000000000
    ///         means that 0.001 of the value being unminted needs to be paid to
    ///         the `VendingMachine` as an unminting fee.
```

Reply:
Please see https://github.com/keep-network/tbtc-v2/pull/9/commits/26858bcf2ea463b9989aef1c57588c25d1b9685f and let me know what you think.